include/zephyr/bluetooth/bluetooth.h

```diff
@@ -685,6 +685,28 @@ enum {
 	 * @note Requires @ref BT_LE_ADV_OPT_USE_NAME
 	 */
 	BT_LE_ADV_OPT_FORCE_NAME_IN_AD = BIT(18),
+
+	/**
+	 * @brief Use new random address on every connectable advertising
+	 * start.
+	 *
+	 * By default, when starting advertising using the @ref
+	 * BT_LE_ADV_OPT_CONNECTABLE option and @kconfig{CONFIG_BT_PRIVACY} is
+	 * disabled, the advertiser set uses the identity address as the
+	 * advertising address.
+	 * When this option is enabled together with @ref
+	 * BT_LE_ADV_OPT_CONNECTABLE, the advertiser set will regenerate
+	 * the advertising address on every advertising start.
+	 * If this option gets disabled, the advertiser set starts using the
+	 * identity address back again.
+	 *
+	 * @note Requires @ref BT_LE_ADV_OPT_CONNECTABLE.
```

Comment:
What happens if this is set and `BT_LE_ADV_OPT_CONNECTABLE` is not set? Rejected? 

Above you write `When this option is enabled together with @ref BT_LE_ADV_OPT_CONNECTABLE` making it sound like it's not mandatory to be used with `BT_LE_ADV_OPT_CONNECTABLE`

Reply:
> What happens if this is set and `BT_LE_ADV_OPT_CONNECTABLE` is not set? Rejected?

Nothing, but I can add check for this.

> Above you write `When this option is enabled together with @ref BT_LE_ADV_OPT_CONNECTABLE` making it sound like it's not mandatory to be used with `BT_LE_ADV_OPT_CONNECTABLE`

That is why I wrote `Requires @ref BT_LE_ADV_OPT_CONNECTABLE.` sentence (like in some other options here).

Reply:
> That is why I wrote Requires @ref BT_LE_ADV_OPT_CONNECTABLE. sentence (like in some other options here).

Right, but my point was that `When this option is enabled together with @ref BT_LE_ADV_OPT_CONNECTABLE` and `Requires @ref BT_LE_ADV_OPT_CONNECTABLE` indicate two different things (one optional and one mandatory requirement), which I found confusing. 

Since there is a strict requirement, `When this option is enabled together with @ref BT_LE_ADV_OPT_CONNECTABLE` can be removed, as it can never not be enabled without `BT_LE_ADV_OPT_CONNECTABLE`

Reply:
@Thalley @PavelVPV would https://github.com/zephyrproject-rtos/zephyr/pull/60700 fit the bill?
It doesn't change current behavior, and makes it explicit that an NRPA will be used (ie not just `RANDOM`).

Reply:
> @Thalley @PavelVPV would #60700 fit the bill? It doesn't change current behavior, and makes it explicit that an NRPA will be used (ie not just `RANDOM`).

In my first version of PR the option was called `..._NRPA` but I changed because per my understanding the static random address (which is taken from identity) can also be NRPA. So I though that calling this option `_RANDOM` (or perhaps I should've call it `_RANDOM_ADDR` or `_RANDOMIZE_ADDR`) makes more clear what option does.

In both, this PR and in #60700, if use the new option in bsim tests that use connection (e.g. audio test), the tests start failing in pairing with reason 0xB which makes me think that these changes aren't enough.

Considering [this](https://github.com/zephyrproject-rtos/zephyr/pull/27907#discussion_r480180674) discussion, @Thalley 's [comment](https://github.com/zephyrproject-rtos/zephyr/pull/60456#discussion_r1270539542) makes me think that if `BT_LE_ADV_OPT_USE_IDENTITY` option is not set, the address should always be randomized.

Reply:
> In both, this PR and in #60700, if use the new option in bsim tests that use connection (e.g. audio test), the tests start failing in pairing

which ones in particular? Also, why would we support that use-case? Pairing with a device using an NRPA seems esoteric.

Reply:
> which ones in particular? 

All.

> Also, why would we support that use-case? Pairing with a device using an NRPA seems esoteric.

I don't know, probably, you are right. Perhaps it should be handled somewhere before pairing is requested?

Reply:
fixed it. could you try it #60700 with mesh's use-case?

Reply:
Works as I need! Thanks!